sdk/spring/azure-spring-boot-core/src/main/java/com/azure/spring/identity/SpringCredentialBuilderBase.java

```diff
@@ -75,4 +80,10 @@ protected String getPropertyValue(String prefix, String propertyKey) {
                      .orElse(null);
     }
 
+    protected <C> C getPropertyValue(Class<C> type, String prefix, String propertyKey, C defaultValue) {
+        return Binder.get(this.environment)
+                     .bind(prefix + propertyKey, type)
+                     .orElse(defaultValue);
```

Comment:
I have a question: Can we use `environment.getProperty(...)` instead?

Reply:
How about do like this:
```


    protected String getPropertyValue(String propertyKey) {
        return getPropertyValue(propertyKey, null);
    }

    protected String getPropertyValue(String propertyKey, String DefaultValue) {
        return Binder.get(this.environment)
                     .bind(propertyKey, String.class)
                     .orElse(null);
    }

...


        String tenantId = getPropertyValue(prefix + "tenant-id");
        String clientId = getPropertyValue(prefix + "client-id");
        String clientSecret = getPropertyValue(prefix + "client-secret");
        String authorityHost = getPropertyValue(prefix + "authority-host", DEFAULT_AUTHORITY_HOST);
```

Reply:
@yiliuTo , seems there is some duplicated work with @saragluna 's PR: https://github.com/Azure/azure-sdk-for-java/pull/22472